File: Dataset/Leetcode/train/125/190.cpp
 class Solution {
public:
    bool XXX(string s) {
        int len = s.size();
        string buf;
        for(int i = 0; i < len; i++)
        {
            int ascii = (int)s[i];
            if(ascii >= 48 && ascii <= 57) buf.push_back(s[i]);
            else if(ascii >= 97 && ascii <= 122) buf.push_back(s[i]);
            else if(ascii >= 65 && ascii <= 90) buf.push_back((char)(ascii + 32));
        }
        len = buf.size();
        for(int l = 0, r = len - 1; l <= r; l++, r--)
        {
            if(buf[l] != buf[r]) return false;
        }
        return true;
    }
};

File: Dataset/Leetcode/train/7/190.java
class Solution {
    public int XXX(int x) {
        if (x == 0) {
            return 0;
        }
        boolean flag = false;
        if (x < 0) {
            flag = true;
            x = -1 * x;
        }
        // 现在x是正值
        long sum = 0;
        while(x > 0) {
            int p = x % 10;
            x = x / 10;
            
            // 这样 123 就被拆成了 p=3, v=12
            sum = sum * 10 + p;
        }
        if (flag) {
            sum = -1l * sum;
        }
        
        if(sum > Integer.MAX_VALUE || sum < Integer.MIN_VALUE) {
            return 0;
        }
        
        return new Long(sum).intValue();   
    }
}

